Doc/faq/design.rst

```diff
@@ -499,8 +499,7 @@ using the :func:`hash` built-in function.  The hash code varies widely depending
 on the key and a per-process seed; for example, "Python" could hash to
 -539294296 while "python", a string that differs by a single bit, could hash
 to 1142331976.  The hash code is then used to calculate a location in an
-internal array where the value will be stored.  Assuming that you're storing
-keys that all have different hash values, this means that dictionaries take
+internal array where the value will be stored.  Hash tables, and therefore dictionaries, take
 constant time -- O(1), in Big-O notation -- to retrieve a key.
```

Comment:
Sooner or later, [someone will complain](https://stackoverflow.com/questions/9214353/hash-table-runtime-complexity-insert-search-and-delete) that the new docs are incorrect because dictionaries have O(n) worst case performance., so some sort of qualifier is still necessary.  We get O(1) performance in the typical case when the hash values are evenly distributed; however, it is easy to construct degenerate cases with a catastrophic pile-ups at a single value.   

Consider using the [wikipedia wording](https://en.wikipedia.org/wiki/Hash_table) where we mention both the average case and worst case.